libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -623,6 +627,54 @@ IotMqttError_t _IotMqtt_deserializeSubackWrapper( _mqttPacket_t * pSuback )
     /* Deserializing SUBACK packet received from the network. */
     managedMqttStatus = MQTT_DeserializeAck( &pIncomingPacket, &( pSuback->packetIdentifier ), NULL );
     status = convertReturnCode( managedMqttStatus );
+
+    /* Remove rejected subscription as the MQTT LTS library do not remove them, it has to handled in shim */
+    if( status == IOT_MQTT_SERVER_REFUSED )
+    {
+        /* Iterate through each status byte in the SUBACK packet. */
+        for( i = 0; i < remainingLength - sizeof( uint16_t ); i++ )
+        {
+            /* Read a single status byte in SUBACK. */
+            subscriptionStatus = *( pVariableHeader + sizeof( uint16_t ) + i );
+
+            /* MQTT 3.1.1 defines the following values as status codes. */
+            switch( subscriptionStatus )
+            {
+                case 0x00:
+                case 0x01:
+                case 0x02:
```

Comment:
please avoid magic numbers and add #define macro and use these numbers.

Reply:
Addressed